### PR TITLE
Publish speclets, clean C# guide titles and title suffix metadata

### PR DESCRIPTION
**Pull request by BillWagner**
Publish the two speclets that are available in preview (lock object, and and params collections).

In addition, clean up the title and title suffixes throughout the C# Guide.

- Remove old title suffixes in individual articles. These included:
  - (C#)
  - C#
  - C# Programming Guide
  - C# [Rr]eference
  - "in C#"

Clean other titles so that C# appears only once in the final built pages.

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/advanced-topics/interface-implementation/default-interface-methods-versions.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/advanced-topics/interface-implementation/default-interface-methods-versions.md) | [Tutorial: Update interfaces with default interface methods](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interface-implementation/default-interface-methods-versions?branch=pr-en-us-40937) |
| [docs/csharp/advanced-topics/interop/how-to-access-office-interop-objects.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/advanced-topics/interop/how-to-access-office-interop-objects.md) | [How to access Office interop objects](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/how-to-access-office-interop-objects?branch=pr-en-us-40937) |
| [docs/csharp/advanced-topics/interop/how-to-use-named-and-optional-arguments-in-office-programming.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/advanced-topics/interop/how-to-use-named-and-optional-arguments-in-office-programming.md) | ["How to use named and optional arguments in Office programming"](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/how-to-use-named-and-optional-arguments-in-office-programming?branch=pr-en-us-40937) |
| [docs/csharp/advanced-topics/interop/walkthrough-creating-and-using-dynamic-objects.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/advanced-topics/interop/walkthrough-creating-and-using-dynamic-objects.md) | [Walkthrough: Creating and Using Dynamic Objects in C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/walkthrough-creating-and-using-dynamic-objects?branch=pr-en-us-40937) |
| [docs/csharp/advanced-topics/interop/walkthrough-office-programming.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/advanced-topics/interop/walkthrough-office-programming.md) | [docs/csharp/advanced-topics/interop/walkthrough-office-programming](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interop/walkthrough-office-programming?branch=pr-en-us-40937) |
| [docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes.md) | [How to create a C/C++ union by using attributes in C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/how-to-create-a-c-cpp-union-by-using-attributes?branch=pr-en-us-40937) |
| [docs/csharp/asynchronous-programming/async-scenarios.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/asynchronous-programming/async-scenarios.md) | [Asynchronous programming scenarios ](https://review.learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/async-scenarios?branch=pr-en-us-40937) |
| [docs/csharp/asynchronous-programming/index.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/asynchronous-programming/index.md) | [Asynchronous programming with async and await](https://review.learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/index?branch=pr-en-us-40937) |
| [docs/csharp/asynchronous-programming/using-async-for-file-access.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/asynchronous-programming/using-async-for-file-access.md) | [Asynchronous file access (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/using-async-for-file-access?branch=pr-en-us-40937) |
| [docs/csharp/delegates-overview.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/delegates-overview.md) | [docs/csharp/delegates-overview](https://review.learn.microsoft.com/en-us/dotnet/csharp/delegates-overview?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/coding-style/coding-conventions.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/coding-style/coding-conventions.md) | [Common C# code conventions](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/coding-style/identifier-names.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/coding-style/identifier-names.md) | ["Identifier names - rules and conventions"](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/exceptions/exception-handling.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/exceptions/exception-handling.md) | [Exception Handling (C# Programming Guide)](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/exception-handling?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/functional/pattern-matching.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/functional/pattern-matching.md) | [docs/csharp/fundamentals/functional/pattern-matching](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/object-oriented/index.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/object-oriented/index.md) | [Overview of object oriented techniques in C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/index?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/object-oriented/inheritance.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/object-oriented/inheritance.md) | [Inheritance - derive types to create more specialized behavior](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/inheritance?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/program-structure/index.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/program-structure/index.md) | ["General Structure of a Program"](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/index?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/tutorials/classes.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/tutorials/classes.md) | [Explore object oriented programming with classes and objects](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/tutorials/classes?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/tutorials/inheritance.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/tutorials/inheritance.md) | [docs/csharp/fundamentals/tutorials/inheritance](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/tutorials/inheritance?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/tutorials/oop.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/tutorials/oop.md) | [Object-Oriented programming (C#)](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/tutorials/oop?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/types/index.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/types/index.md) | [The C# type system](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/index?branch=pr-en-us-40937) |
| [docs/csharp/fundamentals/types/records.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/fundamentals/types/records.md) | ["Record types"](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/records?branch=pr-en-us-40937) |
| [docs/csharp/how-to/compare-strings.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/how-to/compare-strings.md) | [How to compare strings in C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/how-to/compare-strings?branch=pr-en-us-40937) |
| [docs/csharp/how-to/concatenate-multiple-strings.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/how-to/concatenate-multiple-strings.md) | [How to concatenate multiple strings (C# Guide)](https://review.learn.microsoft.com/en-us/dotnet/csharp/how-to/concatenate-multiple-strings?branch=pr-en-us-40937) |
| [docs/csharp/how-to/index.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/how-to/index.md) | [docs/csharp/how-to/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/how-to/index?branch=pr-en-us-40937) |
| [docs/csharp/how-to/modify-string-contents.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/how-to/modify-string-contents.md) | [How to modify string contents in C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/how-to/modify-string-contents?branch=pr-en-us-40937) |
| [docs/csharp/how-to/parse-strings-using-split.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/how-to/parse-strings-using-split.md) | [How to separate strings using String.Split in C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/how-to/parse-strings-using-split?branch=pr-en-us-40937) |
| [docs/csharp/how-to/search-strings.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/how-to/search-strings.md) | ["How to search strings "](https://review.learn.microsoft.com/en-us/dotnet/csharp/how-to/search-strings?branch=pr-en-us-40937) |
| [docs/csharp/language-reference/attributes/caller-information.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/language-reference/attributes/caller-information.md) | [Determine caller information using attributes interpreted by the C# compiler](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/caller-information?branch=pr-en-us-40937) |
| [docs/csharp/language-reference/attributes/general.md](https://github.com/dotnet/docs/blob/09ccf6665fb2246590fcaeb721938f8b1fa733a2/docs/csharp/language-reference/attributes/general.md) | [docs/csharp/language-reference/attributes/general](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/general?branch=pr-en-us-40937) |

</details>

> **Note**
> This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/40937/checks).

### CODE REVIEW

**Comment by BillWagner** [comment]
@gewarren 

This is ready for review. Most of the warnings will be fixed by dotnet/csharplang#8129.

I'm looking at the others to figure out why they appeared suddenly.

File: docs/csharp/language-reference/keywords/internal.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "internal - C# Reference"
-title: "internal - C# Reference"
+title: "internal "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/internal.md:3]:
```suggestion
title: "internal keyword"
```

File: docs/csharp/language-reference/keywords/into.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "into - C# Reference"
-title: "into - C# Reference"
+title: "into "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/into.md:3]:
```suggestion
title: "into keyword"
```

File: docs/csharp/language-reference/keywords/new.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "new - C# Reference"
-title: "new - C# Reference"
+title: "new "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/new.md:3]:
```suggestion
title: "new keyword"
```

File: docs/csharp/language-reference/keywords/private-protected.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "private protected - C# Reference"
-title: "private protected - C# Reference"
+title: "private protected "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/private-protected.md:3]:
```suggestion
title: "private protected keyword"
```

File: docs/csharp/language-reference/keywords/protected-internal.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "protected internal - C# Reference"
-title: "protected internal - C# Reference"
+title: "protected internal "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/protected-internal.md:3]:
```suggestion
title: "protected internal keyword"
```

File: docs/csharp/language-reference/keywords/virtual.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "virtual - C# Reference"
-title: "virtual - C# Reference"
+title: "virtual "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/virtual.md:3]:
```suggestion
title: "virtual keyword"
```

File: docs/csharp/language-reference/keywords/volatile.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ---
 description: "volatile - C# Reference"
-title: "volatile - C# Reference"
+title: "volatile "
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/volatile.md:3]:
```suggestion
title: "volatile keyword"
```

File: docs/csharp/programming-guide/classes-and-structs/constructors.md [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Constructors - C# programming guide"
+title: "Constructors  programming guide"
```
Comment (gewarren) [inline docs/csharp/programming-guide/classes-and-structs/constructors.md:2]:
```suggestion
title: "Constructors"
```

File: docs/csharp/tutorials/exploration/interpolated-strings-local.md [hunk 9]
```diff
@@ -1,5 +1,5 @@
 ---
-title: String interpolation - C# tutorial
+title: String interpolation  tutorial
```
Comment (gewarren) [inline docs/csharp/tutorials/exploration/interpolated-strings-local.md:2]:
```suggestion
title: String interpolation tutorial
```

File: docs/csharp/tutorials/exploration/interpolated-strings.yml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 ### YamlMime:Tutorial
 title: Use string interpolation to construct formatted strings
 metadata:
-  title: String interpolation - C# interactive tutorial
+  title: String interpolation  interactive tutorial
```
Comment (gewarren) [inline docs/csharp/tutorials/exploration/interpolated-strings.yml:4]:
```suggestion
  title: String interpolation interactive tutorial
```

File: docs/csharp/tutorials/records.md [hunk 11]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Use record types - C# tutorial
+title: Use record types  tutorial
```
Comment (gewarren) [inline docs/csharp/tutorials/records.md:2]:
```suggestion
title: Use record types tutorial
```

File: docs/csharp/tutorials/top-level-statements.md [hunk 12]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Top-level statements - C# tutorial
+title: Top-level statements  tutorial
```
Comment (gewarren) [inline docs/csharp/tutorials/top-level-statements.md:2]:
```suggestion
title: Top-level statements tutorial
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**